core/src/main/java/org/apache/calcite/sql/parser/SqlParserUtil.java

```diff
@@ -427,6 +434,11 @@ public static long intervalToMonths(
           + literal, e);
     }
 
+    if (intervalQualifier.timeUnitRange.toString().equals("QUARTER")) {
+      long monthsInQuarter = 3;
+      return ret[0] * ret[2] * monthsInQuarter;
+    }
```

Comment:
Do not use `toString()` to perform comparisons. You already have an enum. Just compare the enum. `TimeUnitRange.QUARTER` exists.

Reply:
Looked into this further and this entire area isn't needed. Since one quarter is 3 months, just perform that translation before calling `fillIntervalValueArray`.